knative-operator/cmd/manager/main.go

```diff
@@ -187,3 +195,49 @@ func setupServerlesOperatorMonitoring(cfg *rest.Config) error {
 	}
 	return nil
 }
+
+// SetupFunctionTasksAndPipelines checks to see if Tekton is installed
+// and if so creates the necessary resources for on-cluster function builds
+func SetupFunctionTasksAndPipelines(cfg *rest.Config) error {
+	cl, err := client.New(cfg, client.Options{})
+	if err != nil {
+		return fmt.Errorf("failed to create a client: %w", err)
+	}
+
+	// Check to see if Tekton is installed by looking for the pipelines CRD
```

Comment:
I suggest we have a common logic for OCP feature discovery.
We check the cluster operator status [here](https://github.com/openshift-knative/serverless-operator/blob/main/knative-operator/cmd/manager/main.go#L176-L187) for console for example.